src/mongo/integration/standalone/dbclient_test.cpp

```diff
@@ -394,6 +394,67 @@ namespace {
         ASSERT_TRUE(saw_capped);
     }
 
+    TEST_F(DBClientTest, EnumerateCollections) {
+        const int n_collections = 9;
+        const int batch_size = 3;
+
+        for (int i = 0; i < n_collections; ++i) {
+            std::stringstream ss;
+            ss << TEST_DB + ".COLL" << i;
+            c.createCollection(ss.str());
+        }
+
+        BSONObjBuilder bob;
+        bob.appendRegex("name", "COLL\\d$");
+        auto_ptr<DBClientCursor> cursor = c.enumerateCollections(TEST_DB, bob.obj(), batch_size);
+
+        if (serverGTE(&c, 2, 8))
+            ASSERT_EQUALS(cursor->getns(), TEST_DB + ".$cmd.listCollections");
+        else
+            ASSERT_EQUALS(cursor->getns(), TEST_DB + ".system.namespaces");
+
+        int coll_count = 0;
+        while (cursor->more()) {
+            // The shims don't support the objsLeftInBatch cursor interface for initial batch.
```

Comment:
I'm not sure that that is OK. We are implementing the interface for DBClientCursor, so... we need to implement the interface?